File: GO/Dama_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestDama(t *testing.T) {
	tables := []struct {
		x1   float64
		y1   float64
		x2   float64
		y2   float64
		want int
	}{
		{4, 4, 6, 2, 1},
		{3, 5, 3, 5, 0},
		{5, 5, 4, 3, 2},
	}

	for _, table := range tables {
		got := minMoviments(table.x1, table.y1, table.x2, table.y2)

		if got != table.want {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}

File: PHP/DamaTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("Dama.php");

class DamaTest extends TestCase {

    public function testDama() {
        $obj = new Dama();
        //Test 1
        $want = 1;
        $got = $obj->minMoviments(4, 4, 6, 2);
        $this->assertEquals($want, $got);
        //Test 2
        $want = 0;
        $got = $obj->minMoviments(3, 5, 3, 5);
        $this->assertEquals($want, $got);
        //Test 3
        $want = 2;
        $got = $obj->minMoviments(5, 5, 4, 3);
        $this->assertEquals($want, $got);
    }

}
